src/clj/harja/palvelin/palvelut/budjettisuunnittelu.clj

```diff
@@ -329,7 +329,8 @@
                                             (fn [{:keys [hoitokausi tavoitehinta kattohinta]}]
                                               (when (and kattohinta
                                                       ;; Sallitaan kattohinnan arvoksi nolla mikäli tavoitehintakin on nolla.
-                                                      (not (zero? kattohinta)) (not (zero? tavoitehinta))
+                                                      (not (and (zero? kattohinta)
+                                                             (zero? tavoitehinta)))
```

Comment:
Mun ymmärtääkseni jos tavoitehinta on nil, niin tämmä kaatuu null pointer exceptioniin.

Eli ehkä kehottaisin pävittämään tuon ehdon esim muotoon:
(fn [kattohinta tavoitehinta hoitokausi]
             (cond (and tavoitehinta kattohinta (zero? tavoitehinta) (zero? kattohinta)) hoitokausi
                   (and tavoitehinta kattohinta (<= kattohinta tavoitehinta)) hoitokausi
                   :else nil))

Reply:
Korjattu, hieman eri tavalla kuin ehdotus. Ehdotuksessa nollat palauttaisivat virheen, se kuitenkin halutaan varmaankin sallia.